lib/IRGen/GenExistential.cpp

```diff
@@ -663,60 +661,52 @@ namespace {
-      IGF.emit##Name##CopyAssign(dest, src, Refcounting); \
-    } \
-    void emitValueInitializeWithCopy(IRGenFunction &IGF, \
-                                     Address dest, Address src) const { \
-      IGF.emit##Name##CopyInit(dest, src, Refcounting); \
-    } \
-    void emitValueAssignWithTake(IRGenFunction &IGF, \
-                                Address dest, Address src) const { \
-      IGF.emit##Name##TakeAssign(dest, src, Refcounting); \
-    } \
-    void emitValueInitializeWithTake(IRGenFunction &IGF, \
-                                    Address dest, Address src) const { \
-      IGF.emit##Name##TakeInit(dest, src, Refcounting); \
-    } \
-    void emitValueDestroy(IRGenFunction &IGF, Address addr) const { \
-      IGF.emit##Name##Destroy(addr, Refcounting); \
-    } \
-    StringRef getStructNameSuffix() const { return "." #name "ref"; } \
-    REF_STORAGE_HELPER(Name, FixedTypeInfo) \
+#define NEVER_LOADABLE_CHECKED_REF_STORAGE(Name, name, ...)                    \
+  class AddressOnly##Name##ClassExistentialTypeInfo final                      \
+      : public AddressOnlyClassExistentialTypeInfoBase<                        \
+            AddressOnly##Name##ClassExistentialTypeInfo, FixedTypeInfo> {      \
+    bool IsOptional;                                                           \
+                                                                               \
+  public:                                                                      \
+    AddressOnly##Name##ClassExistentialTypeInfo(                               \
+        ArrayRef<const ProtocolDecl *> protocols, llvm::Type *ty,              \
+        SpareBitVector &&spareBits, Size size, Alignment align,                \
+        ReferenceCounting refcounting, bool isOptional)                        \
+        : AddressOnlyClassExistentialTypeInfoBase(                             \
+              protocols, refcounting, ty, size, std::move(spareBits), align,   \
+              IsNotPOD, IsNotBitwiseTakable, IsFixedSize),                     \
+          IsOptional(isOptional) {}                                            \
+    TypeLayoutEntry *buildTypeLayoutEntry(IRGenModule &IGM,                    \
+                                          SILType T) const override {          \
+      if (Refcounting == ReferenceCounting::Native) {                          \
+        return IGM.typeLayoutCache.getOrCreateScalarEntry(                     \
+            *this, T, ScalarKind::Native##Name##Reference);                    \
+      } else {                                                                 \
+        return IGM.typeLayoutCache.getOrCreateScalarEntry(                     \
+            *this, T, ScalarKind::Unknown##Name##Reference);                   \
+      }                                                                        \
+    }                                                                          \
```

Comment:
The buildTypeLayoutEntry is the only functional change here, the rest of the macro definition is formatting is sucked in by clang-format.